tools/testsys/src/aws_resources.rs

```diff
@@ -497,21 +591,25 @@ impl AwsEcs {
                 conflicts_with: None,
                 agent: Agent {
                     name: "ecs-provider".to_string(),
-                    image: testsys_images.ecs_resource.clone(),
-                    pull_secret: testsys_images.secret.clone(),
+                    image: testsys_images
+                        .ecs_resource_agent_image
+                        .as_ref()
+                        .cloned()
```

Comment:
Is the `as_ref()` is necessary?
```suggestion
                        .cloned()
```